Newsletters: add 266 (2023-08-30) (#1259)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2023-08-30-newsletter.md

  succeeds, the watchtower can broadcast the decrypted penalty
  transaction.  This protects the user against their counterparty
  publishing a revoked channel state when the user is unavailable.

- [LDK #2411][] and [#2412][ldk #2412] add an API for constructing
  payment paths for [blinded payments][topic rv routing].  The PRs help
  separate LDK's code for [onion messages][topic onion messages] (which
  use blinded paths) from blinded paths themselves.  A follow up in
  [#2413][ldk #2413] will actually add support for route blinding.

- [LDK #2507][] adds a workaround for a longstanding issue in another
  implementation that leads to unnecessary channel force closures.

- [LDK #2478][] adds an event that provides information about a forwarded
  [HTLC][topic htlc] that has now been settled, including what channel
  it came from, the amount of the HTLC, and the amount of fee we
  collected from it.

- [LND #7904][] adds experimental support for "simple taproot
  channels", allowing LN funding and commitment transactions to use
  [P2TR][topic taproot] with support for [MuSig2][topic musig]
  scriptless [multisignature][topic multisignature] signing when both

>   parties are cooperating.  This reduces block space and improves

Reviewer: ```suggestion
  parties are cooperating.  This reduces the transaction weight and improves
```

---
title: 'Bitcoin Optech Newsletter #266'
permalink: /en/newsletters/2023/08/30/
name: 2023-08-30-newsletter
slug: 2023-08-30-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

- **Disclosure of past LN vulnerability related to fake funding:** Matt
  Morehouse [posted][morehouse dos] to the Lighting-Dev mailing list
  with the summary of a vulnerability he had previously [responsibly
  disclosed][topic responsible disclosures] and which is now addressed
  in the latest versions of all popular LN implementations.  To
  understand the vulnerability, imagine that Bob runs an LN node and he
  receives a request from Mallory's node to open a new channel.  They go
  through the channel opening process up to the stage where Mallory is
  supposed to broadcast a transaction that funds the channel.  In order
  to later use that channel, Bob needs to store some state related to it
  and begin scanning new blocks for the transaction to become
  sufficiently confirmed.  If Mallory never broadcasts the transaction,
  Bob's storage and scanning resources are wasted.  If Mallory repeats

>   the process thousands or millions of times, it might waste Bob's

Author: I think "might" is correct here.  This outcome is not certain (and, indeed, CLN managed to avoid it).

  nodes that process gossip must only accept messages that are signed by
  a key associated with a UTXO, otherwise they can be forced into
  processing and relaying spam messages, or attempting to forward
  payments over nonexistent channels (which will always fail).  The new
  built-in `UtxoSource` works for LN nodes connected to a local Bitcoin
  Core.

- [LDK #2337][] makes it easier to use LDK for building
  [watchtowers][topic watchtowers] that run independently from a user's
  wallet but can receive encrypted LN penalty transactions from the
  user's node.  A watchtower can then extract information from each
  transaction in new blocks and use that information to attempt to
  decrypt the previously-received encrypted data.  If the decryption
  succeeds, the watchtower can broadcast the decrypted penalty
  transaction.  This protects the user against their counterparty
  publishing a revoked channel state when the user is unavailable.

- [LDK #2411][] and [#2412][ldk #2412] add an API for constructing
  payment paths for [blinded payments][topic rv routing].  The PRs help
  separate LDK's code for [onion messages][topic onion messages] (which

>   use blinded paths) from blinded paths themselves.  A follow up in

Reviewer 2: ```suggestion
  use blinded paths) from blinded paths themselves.  A follow-up in
```

  generating a [schnorr signature][topic schnorr signatures] nonce for
  protection against [side channel][topic side channels] attacks, yet Bitcoin
  Core doesn't provide auxiliary randomness in its implementation. Andrew Chow
  answers that the current implementation is still safe and that no PR has been
  opened to address the recommendation.

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test release candidates.*

- [Core Lightning 23.08][] is the newest major release of this popular
  LN node implementation.  New features include the ability to
  change several node configuration settings without restarting the
  node, support for [Codex32][topic codex32]-formatted [seed][topic
  bip32] backup and restore, a new experimental plugin for improved
  payment pathfinding, experimental support for [splicing][topic
  splicing], and the ability to pay locally-generated invoices, among
  many other new features and bug fixes.

- [LND v0.17.0-beta.rc1][] is a release candidate for the next major
  version of this popular LN node implementation.  A major new

>   experimenal feature planned for this release, which could likely

Reviewer 3: ```suggestion
  experimental feature planned for this release, which could likely
```
